TESTS/mbed-crypto/sanity/main.cpp

```diff
@@ -87,6 +87,44 @@ void test_crypto_hash_verify(void)
     mbedtls_psa_crypto_free();
 }
 
+void test_crypto_symmetric_cipher_encrypt_decrypt(void)
+{
+    psa_key_slot_t slot = 1;
+    psa_key_type_t key_type = PSA_KEY_TYPE_AES;
+    psa_algorithm_t alg = PSA_ALG_CBC_NO_PADDING;
+    psa_cipher_operation_t operation;
+    psa_key_policy_t policy;
+    unsigned char input[16], iv[16], encrypted[16], decrypted[16];
+    size_t output_len, iv_len, key_bits = 256;
+
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_crypto_init());
+
+    psa_key_policy_init(&policy);
+    psa_key_policy_set_usage(&policy, PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT, alg);
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_set_key_policy(slot, &policy));
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_generate_key(slot, key_type, key_bits, NULL, 0));
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_generate_random(input, sizeof(input)));
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_cipher_encrypt_setup(&operation, slot, alg));
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_cipher_generate_iv(&operation, iv, sizeof(iv), &iv_len));
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_cipher_update(&operation, input, sizeof(input),
+                                                     encrypted, sizeof(encrypted), &output_len));
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_cipher_finish(&operation, encrypted + output_len,
+                                                     sizeof(encrypted) - output_len, &output_len));
+    TEST_ASSERT_EQUAL(PSA_SUCCESS, psa_cipher_abort(&operation));
```

Comment:
We should add another TEST_ASSERT_EQUAL_UINT8_ARRAY here, to check that the encrypted value matches what we expect the data to get encrypted to.

Otherwise, this test would pass if the buffer was never encrypted.

Reply:
I know, but since the key is generated rather than imported I don't see how this can be done. The jira tasks states that the keys should be generated.

Reply:
We have a separate key generation test. Not all tests must use generated keys.

Reply:
I'll change this to use a fixed imported key.